### PR TITLE
Modified local-volume provisioner e2e test to use mounted volumes

### LINKED ISSUE #57004: Need local persistent volume provisioner e2e tests to use mounts points instead of directories
The local PV provisioner is being updated to discover only real mount points (and not directories), that resolves some other related issues. https://github.com/kubernetes-incubator/external-storage/pull/499

The related e2e tests using the local PV provisioner need to be updated accordingly.

### PR DESCRIPTION
**Pull request by ianchakeres**
**What this PR does / why we need it**: We are changing the local volume provisioner to discover only mount points, https://github.com/kubernetes-incubator/external-storage/pull/499. This PR shifts the volumes from directories to mount points for the provisioner e2e tests.

**Which issue(s) this PR fixes**: 
Fixes https://github.com/kubernetes/kubernetes/issues/57004

**Special notes for your reviewer**:

**Release note**:
```
NONE
```

### CODE REVIEW

File: test/e2e/storage/persistent_volumes-local.go [hunk 1]
```diff
@@ -344,11 +347,24 @@ var _ = SIGDescribe("PersistentVolumes-local [Feature:LocalPersistentVolumes] [S
...
 			framework.WaitForControlledPodsRunning(config.client, config.ns, daemonSetName, kind)
 
-			By("Creating a directory under discovery path")
-			framework.Logf("creating local volume under path %q", volumePath)
-			mkdirCmd := fmt.Sprintf("mkdir %v -m 777", volumePath)
-			err := framework.IssueSSHCommand(mkdirCmd, framework.TestContext.Provider, config.node0)
+			By("Creating a directory for content")
+			framework.Logf("creating local directory at path %q", volumeContentDir)
+			mkdirCmd1 := fmt.Sprintf("mkdir %v -m 777", volumeContentDir)
+			mkdirCmd1Err := framework.IssueSSHCommand(mkdirCmd1, framework.TestContext.Provider, config.node0)
```
Comment (copejon) [inline test/e2e/storage/persistent_volumes-local.go]: nit - go idiom would be to move the `err` var declaration here (from L365) and re-assign it for each error catch.

**Comment by copejon** [comment]
The change looks good.  Can you explain what the value added is?  Does using a bind mount improve test performance / stability?

File: test/e2e/storage/persistent_volumes-local.go [hunk 2]
```diff
@@ -325,30 +325,47 @@ var _ = SIGDescribe("PersistentVolumes-local [Feature:LocalPersistentVolumes] [S
...
+
+			By("Creating a directory under discovery path for mounting")
+			framework.Logf("creating local directory at path %q", volumePath)
+			mkdirCmd2 := fmt.Sprintf("mkdir %v -m 777", volumePath)
+			err = framework.IssueSSHCommand(mkdirCmd2, framework.TestContext.Provider, config.node0)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Mounting directory under discovery path")
+			framework.Logf("mounting local directory %q at path %q", volumeContentDir, volumePath)
+			mntCmd := fmt.Sprintf("sudo mount --bind %v %v", volumeContentDir, volumePath)
```
Comment (msau42) [inline test/e2e/storage/persistent_volumes-local.go]: Would it be simpler to just have the content dir == mount dir?

Comment (ianchakeres) [reply test/e2e/storage/persistent_volumes-local.go]: I'll try it out.

Comment (ianchakeres) [reply test/e2e/storage/persistent_volumes-local.go]: @msau42 - PR updated.

File: test/e2e/storage/persistent_volumes-local.go [hunk 3]
```diff
@@ -889,6 +887,30 @@ func cleanupLocalVolumeProvisioner(config *localTestConfig, volumePath string) {
...
+}
+
+func cleanupLocalVolumeProvisionerMountPoint(config *localTestConfig, volumePath string) {
+	By(fmt.Sprintf("Unmounting the test mount point from %q", volumePath))
+	umountCmd := fmt.Sprintf("sudo umount %v", volumePath)
+	err := framework.IssueSSHCommand(umountCmd, framework.TestContext.Provider, config.node0)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Removing the test discovery directory")
+	removeCmd2 := fmt.Sprintf("rm -r %s", path.Join(hostBase, discoveryDir))
```
Comment (msau42) [inline test/e2e/storage/persistent_volumes-local.go]: This is already done at L879, right?

File: test/e2e/storage/persistent_volumes-local.go [hunk 4]
```diff
@@ -889,6 +886,30 @@ func cleanupLocalVolumeProvisioner(config *localTestConfig, volumePath string) {
...
+}
+
+func cleanupLocalVolumeProvisionerMountPoint(config *localTestConfig, volumePath string) {
+	By(fmt.Sprintf("Unmounting the test mount point from %q", volumePath))
+	umountCmd := fmt.Sprintf("sudo umount %v", volumePath)
+	err := framework.IssueSSHCommand(umountCmd, framework.TestContext.Provider, config.node0)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Removing the test mount point")
+	removeCmd := fmt.Sprintf("rm -r %s", path.Join(hostBase, volumePath))
```
Comment (msau42) [inline test/e2e/storage/persistent_volumes-local.go]: How come this isn't just volumePath like the unmount?

Comment (ianchakeres) [reply test/e2e/storage/persistent_volumes-local.go]: Updated. Thanks for saving me a test cycle.

**Pull request merged**